lms/templates/dates_banner.html

```diff
@@ -17,76 +17,89 @@
-                        <input type="hidden" name="reset_deadlines_redirect_url_base" value="${reset_deadlines_redirect_url_base}">
-                        <input type="hidden" name="reset_deadlines_redirect_url_id_dict" value="${reset_deadlines_redirect_url_id_dict}">
-                        <button class="btn reset-deadlines-button">${_("Reset my deadlines")}</button>
-                    </form>
-                % endif:
-                % if (on_dates_tab_as_audit and has_locked_assignments) or on_course_outline_page_as_audit or (on_courseware_page_as_audit and missed_deadlines):
-                    <a href="${verified_upgrade_link}">
-                        <button type="button">
-                            % if on_dates_tab:
-                                % if missed_deadlines:
-                                    ${_('Upgrade to shift due dates')}
-                                % else:
-                                    ${_('Upgrade now')}
-                                % endif
-                            % else:
-                                ${_('Upgrade to shift due dates')}
-                            % endif
-                        </button>
-                    </a>
-                % endif
+                <a href="${verified_upgrade_link}">
+                    <button type="button">
+                        ${_('Upgrade to shift due dates')}
+                    </button>
+                </a>
             </div>
         % endif
     </div>
+</%def>
+<%def name="upgrade_to_complete_graded_banner()">
+    <div class="dates-banner ${additional_styling_class}">
+        <div class="dates-banner-text">
+            % if is_mobile_app:
+                <strong>${_('You are auditing this course,')}</strong>
+                ${_(' which means that you are unable to participate in graded assignments.')}
+                ${_('Graded assignments are available to Verified Track learners.')}
+            % else:
+                <strong>${_('You are auditing this course,')}</strong>
+                ${_(' which means that you are unable to participate in graded assignments.')}
+                ${_(' To complete graded assignments as part of this course, you can upgrade today.')}
+            % endif
+        </div>
+        % if not is_mobile_app:
+            <div class="upgrade-button">
+                <a href="${verified_upgrade_link}">
+                    <button type="button">
+                        ${_('Upgrade now')}
+                    </button>
+                </a>
+            </div>
+        % endif
+    </div>
+</%def>
+% if on_dates_tab and content_type_gating_enabled and not missed_deadlines:
+    ${upgrade_to_complete_graded_banner()}
+% elif missed_deadlines:
+    % if content_type_gating_enabled:
+        ${upgrade_to_reset_banner()}
+    % else:
+        ${reset_dates_banner()}
+    % endif
 % endif
```

Comment:
1. `missed_deadlines` gets set in the outline fragment here: https://github.com/edx/edx-platform/pull/24085/files#diff-368199a738c593327eeb4c86bffa9470R33. I put it there because that same template is also what does the javascript manipulation to show the page.

2. `content_type_gating_enabled` will be `False` for any users that aren't audit in an audit/verified course. In those cases, we'll let users in other course types reset their deadlines (per @Ahernkb's direction)

Reply:
1. great didn't see that change there
2. sounds good